docs/hugo/content/design/ADR-2023-05-Installing-Only-Selected-CRDs.md

```diff
@@ -0,0 +1,248 @@
+
+The user would need re-specify what resources they want every time they upgrade
+
+Pros:
+* Easy for us
+
+Cons:
+* Very easy to make a mistake and accidentally leave a resource off they want.
+
+#### Answer 3b: Use a configuration technique that persists across upgrades
+
+Pros:
+* Satisfies the requirement.
+
+Cons:
+* Would augment/complicate the approach we already have
+* More complex for us to implement, especially with Helm. Helm owns any resource it creates, 
+  and deletes it if not specified on the next upgrade. See [Using a ConfigMap](#answer-2b--use-a-configmap) for some 
+  workarounds.
+
+#### Answer 3c: Query existing CRDs and always support those
+
+Combined with some form of non-persistent storage such as [command line](#answer-2a--using-the-existing-command-line-parameter)
+we would query all existing CRDs and install/upgrade them, while adding whatever new CRDs were specified in 
+the non-persistent storage.
+
+Pros:
+* Automatically supports seamless upgrades.
+* Installed resources are always upgrade to the latest version, even if user has forgotten about them.
+* Could be built onto existing cmdline configuration.
+* Hard for users to get wrong.
+
+Cons:
+* ???
+
+## Decision
+
+### Answer to [question #1: What to install by default?](#question-1--what-to-install-by-default)
+
+A combination of  [Answer 1b: Nothing by default](#answer-1b--nothing-by-default) and 
+[Answer 1c: Mandatory configuration](#answer-1c--mandatory-configuration).
+
+Specifically, the operator pod should exit with an error if there are no CRDs configured in `--crd-pattern` _and_ no
+existing CRDs in the cluster. If a `--crd-pattern` is specified, or if there are already existing ASO CRDs in the cluster
+then the pod will launch without an error, install/upgrade the required CRDs and then start watches on those CRDs.
+
+### Answer to [question #2: How do users specify what CRDs to install?](#question-2--how-do-users-specify-what-crds-to-install)
+
+[Answer 2a: Using the existing command line parameter](#answer-2a--using-the-existing-command-line-parameter)
+
+This parameter will come to mean: "New CRDs to install to the cluster".
+
+### Answer to [question #3: How do we ensure consistent configuration as ASO is upgraded?](#question-3--how-do-we-ensure-consistent-configuration-as-aso-is-upgraded)
+
+[Answer 3c: Query existing CRDs and always support those](#answer-3c--query-existing-crds-and-always-support-those)
+
+### Open questions and their answers
+
+#### Should we do "graph-based" inclusions of resources?
+
+If a user imports `microsoft.storage/storageAccountsBlobService` should we automatically determine that they 
+also want `microsoft.storage/storageAccounts` since that's the `owner` type?
+
+Conclusion: No, users must list all resources they want explicitly. We strongly recommend using full provider imports such as 
+`microsoft.networking/*` rather than tightly scoping to specific resources.
+
+## Status
+
+Proposed.
```

Comment:
Open question, even without a predefined set of "default" resources, do we want a "default" set in the ASO install command on the readme so that people know what to do?

Consensus seems to be: yes.

Reply:
Made a note of this. Docs will be in the implementation PR